test/rich-text/plugins/link-paste-handler.test.ts

```diff
@@ -0,0 +1,145 @@
+import { linkPasteHandler } from "../../../src/rich-text/plugins/link-paste-handler";
+import "../../matchers";
+import {
+    applySelection,
+    cleanupPasteSupport,
+    createState,
+    createView,
+    dispatchPasteEvent,
+    setupPasteSupport,
+} from "../test-helpers";
+
+const nonURLTestData = [
+    "not a URL",
+    "https://example.org but not by itself",
+    "URL https://example.org in the middle",
+];
+
+const URLTestData = [
+    "https://example.org",
+    "https://example.org/wiki/BestSubjectEver",
+    "https://sub.complicated.domain:8080/path/to/whatever.png#hash?query=parameter%20test",
+];
+
+describe("linkPasteHandler plugin", () => {
+    beforeAll(setupPasteSupport);
+    afterAll(cleanupPasteSupport);
+
+    it.each(nonURLTestData)(
+        "should handle pasting non-URL text (%#)",
+        (text) => {
+            const view = createView(createState("", [linkPasteHandler]));
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const insertedNode = view.state.doc.nodeAt(
+                view.state.selection.from - 1
```

Comment:
optional: This can be a bit finicky, as any changes to how this/other plugins handle cursors/pasting will break these tests. You can use my _really ugly_ (but effective) `expect(...).toMatchNodeTree(...)` method to check the contents of the document ([example](https://github.com/StackExchange/Stacks-Editor/blob/main/test/rich-text/plugins/code-paste-handler.test.ts#L106))

Reply:
My original approach here was to do that, but I ran into issues with checking for marks in expected output. I don't recall the specific errors I was hitting, but I spent a fair amount of time staring at the object printout in the browser and matching it with my test setup... and just couldn't get it right. Since you marked it optional, I'd prefer to leave these as is at least for now.

Reply:
No prob! I bet I know what you ran into too. The way the matcher is currently written, if you include the value, the entire value needs to match. In the case of arrays, this includes all items and their values. Doing something [like this](https://github.com/StackExchange/Stacks-Editor/blob/main/test/shared/markdown-parser.test.ts#L39) should work. That being said, it is ok to leave this as-is.

Reply:
Ooh yeah, that looks promising. In my attempts, I thought I had the entire tree written out, but I suspect what the browser was showing me was trimmed down.